extractors/postgres/extractor_test.go

```diff
@@ -0,0 +1,236 @@
+		db, err := sql.Open("postgres", "postgres://root:pass@localhost:5432?sslmode=disable")
+		if err != nil {
+			db.Close()
+			t.Fatal(err)
+		}
+		err = mockDataGenerator(db)
+		if err != nil {
+			db.Close()
+			t.Fatal(err)
+		}
+		db.Close()
+		db, err = sql.Open("postgres", "postgres://root:pass@localhost:5432?sslmode=disable")
+		if err != nil {
+			db.Close()
+			t.Fatal(err)
+		}
+		result, err := extractor.Extract(map[string]interface{}{
+			"user_id":       user,
+			"password":      pass,
+			"host":          "localhost:5432",
+			"database_name": testDB,
+		})
+		if err != nil {
+			cleanDatabase(db)
+			db.Close()
+			t.Fatal(err)
+		}
+		cleanDatabase(db)
+		db.Close()
+		expected := getExpectedVal()
+		assert.Equal(t, result, expected)
+	})
+}
+
+func getExpectedVal() (expected []map[string]interface{}) {
+	expected = []map[string]interface{}{
+		{
+			"columns": []map[string]interface{}{
+				{
+					"data_type":   "integer",
+					"field_name":  "applicant_id",
+					"is_nullable": "YES",
+					"length":      0,
+				},
+				{
+					"data_type":   "character varying",
+					"field_name":  "first_name",
+					"is_nullable": "YES",
+					"length":      255,
+				},
+				{
+					"data_type":   "character varying",
+					"field_name":  "last_name",
+					"is_nullable": "YES",
+					"length":      255,
+				},
+			},
+			"database_name": "mockdata_meteor_metadata_test",
+			"table_name":    "applicant",
+		},
+		{
+			"columns": []map[string]interface{}{
+				{
+					"data_type":   "character varying",
+					"field_name":  "department",
+					"is_nullable": "YES",
+					"length":      255,
+				},
+				{
+					"data_type":   "character varying",
+					"field_name":  "job",
+					"is_nullable": "YES",
+					"length":      255,
+				},
+				{
+					"data_type":   "integer",
+					"field_name":  "job_id",
+					"is_nullable": "YES",
+					"length":      0,
+				},
+			},
+			"database_name": "mockdata_meteor_metadata_test",
+			"table_name":    "jobs",
+		},
+	}
+	return
+}
+
+func mockDataGenerator(db *sql.DB) (err error) {
+	_, err = db.Exec(fmt.Sprintf(`DROP ROLE IF EXISTS "%s";`, user))
+	if err != nil {
+		return
+	}
+	_, err = db.Exec(fmt.Sprintf(`CREATE ROLE "%s" WITH SUPERUSER LOGIN PASSWORD '%s';`, user, pass))
+	if err != nil {
+		return
+	}
+	_, err = db.Exec(fmt.Sprintf(`SET ROLE "%s";`, user))
+	if err != nil {
+		return
+	}
+	_, err = db.Exec(fmt.Sprintf("DROP DATABASE IF EXISTS %s", testDB))
+	if err != nil {
+		return
+	}
+	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", testDB))
+	if err != nil {
+		return
+	}
+	db.Close()
+	db, err = sql.Open("postgres", fmt.Sprintf("postgres://%s:%s@localhost:5432/%s?sslmode=disable", user, pass, testDB))
+	if err != nil {
+		return
+	}
+	table1 := "applicant"
+	var1 := "(applicant_id int, last_name varchar(255), first_name varchar(255))"
+	table2 := "jobs"
+	var2 := "(job_id int, job varchar(255), department varchar(255))"
+	tableQuery := "CREATE TABLE "
```

Comment:
CMIIW but I think we could move this to `createTable` and since it will only be used in that function.

Reply:
also for readability maybe we could do this
```
err = createTable(db, "applicant",  columnsQuery)  // columnsQuery is var1
```